system/Model.php

```diff
@@ -1247,66 +1209,55 @@ public function purgeDeleted()
 		}
 
 		return $this->builder()
-				->where($this->table . '.' . $this->deletedField . ' IS NOT NULL')
-				->delete();
+			->where($this->table . '.' . $this->deletedField . ' IS NOT NULL')
+			->delete();
 	}
 
-	//--------------------------------------------------------------------
-
 	/**
 	 * Sets $useSoftDeletes value so that we can temporarily override
 	 * the softdeletes settings. Can be used for all find* methods.
 	 *
 	 * @param boolean $val
 	 *
-	 * @return Model
+	 * @return $this
 	 */
-	public function withDeleted($val = true)
+	public function withDeleted(bool $val = true)
```

Comment:
This is the only potential break I see, for anyone passing in non-booleans , but since we have the doc block I think it is safe.

Reply:
I don't see it as breaking since the user guide did not mention it to be accepting a required argument, albeit having a default value. If someone is using a reputable IDE the doc block will surely inform that it is accepting booleans as param.